scripts/cmake/CompilerSetup.cmake

```diff
@@ -52,7 +52,11 @@ if(COMPILER_IS_GCC)
             set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_GLIBCXX_DEBUG -D_GLIBCXX_DEBUG_ASSERT -D_GLIBCXX_DEBUG_PEDASSERT -D_GLIBCXX_DEBUG_VERIFY")
         endif()
     endif()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CPU_FLAGS} -std=c++11 -Wno-deprecated -Wall -Wextra")
+	if(CYGWIN)
+		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CPU_FLAGS} -std=gnu++0x -Wa,-mbig-obj -Wno-deprecated -Wall -Wextra")
```

Comment:
In cases where something goes wrong for some reasons under some circumstances a version/exact error message/locations etc. are helpful.
 - What is the version of g++
 - Which classes are heavy templated
 - Where are those classes embedded (which library or obj-file)
 - Which limits of ccache are exceeded
???

Also why there is `-std=c++0x` instead of `-std=c++11`?